doc/modules/mod_event_pusher_push.md

```diff
@@ -1,57 +1,163 @@
-Any publish error may result in disabling push notifications to this node.
+It provides push notification data to the service that delivers actual notification to a client device.
 
-### Options
+We've prepared [a detailed tutorial](../user-guide/Push-notifications.md) about a proper push notifications setup both on a client and a server side.
+
+Please make sure that clients provide all form fields required by the specified `PubSub` node.
+Some publish errors may result in disabling push notifications for the specific device until it attempts to enable them again.
+
+## Options
 
 * **backend** (atom, default: `mnesia`) - Backend to use for storing the registrations.
  Possible options are `mnesia` and `rdbms`.
 * **wpool** (list, default: `[]`) - List of options that will be passed to the `worker_pool` library that handles all the requests.
  Please refer to the [Project Site](https://github.com/inaka/worker_pool) for more details.
 * **plugin_module** (atom, default: `mod_event_pusher_push_plugin_defaults`) - module implementing `mod_event_pusher_push_plugin` behaviour,
   used for dynamic configuration of push notifications. Read more about it [here](#plugin-module)
+* **virtual_pubsub_hosts** (list of strings, default: `[]`) - a list of "simulated" Publish-Subscribe domains. See [relevant section](#virtual-pubsub-hosts) for more details. You may use `@HOST@` pattern in the domain name. It will automatically be replaced by a respective XMPP domain (e.g. `localhost`)
 
-### Plugin module
+## Virtual PubSub hosts
 
-A plugin module handles dynamic configuration of push notifications. It implements `mod_event_pusher_push_plugin` behaviour which
-requires two callbacks:
+If a notification is published to one of the configured domains, the internal push notifications hook is executed in MongooseIM instead of the XEP-0357 typical behaviour.
+If an existing PubSub domain is added to this list, it will be shadowed in the push notifications context.
+It enables easy migration from PubSub-ful deployments to PubSub-less variants.
 
-* `should_publish/3` - callback used for filtering push notifications. A push notification is triggered for given a message only if this
-callback returns `true`.
+### Advantages
 
-```
--spec should_publish(From :: ejabberd:jid(), To :: ejabberd:jid(), Packet :: jlib:xmlel()) -> boolean().
+* No need to use PubSub at all
+* Slightly more efficient (PubSub has noticable impact on heavily loaded systems)
+* Simpler client-side usage
+
+### Drawbacks
+
+* Not suitable for public deployments, which allow connections from 3rd party client applications. Notifications won't work at all in such case.
+* It is only partially XEP-0357 compliant, so 3rd party clients most probably won't work even in private deployments.
+* This option works out of the box only with the default plugin. If you create a custom one, it must take the virtual hosts option into account as well.
```

Comment:
```suggestion
* This option works out of the box only with the default plugin. If you create a custom one, it must take the virtual host option into account as well.
```

not sure if it needs the change, if I understand correctly the intention is in fact "take the 'virtual hosts' option"?

Reply:
the option name is virtual_pubsub_hosts